### PR TITLE
[ci] Add experimental feature in tests and cargo check

### PR DESCRIPTION
**Pull request by alvicsam**

### CODE REVIEW

File: .github/workflows/check-labels.yml [hunk 1]
```diff
@@ -33,8 +33,8 @@ jobs:
           # Temporary, before https://github.com/paritytech/labels/pull/29 is not merged
           git clone https://github.com/paritytech/labels
           cd labels
-          git fetch origin the-right-joyce-monorepo-labels
-          git checkout the-right-joyce-monorepo-labels
+          git fetch origin main
+          git checkout main
           cd ..
```
Comment (ggwpez) [inline .github/workflows/check-labels.yml:38]: `git clone https://github.com/paritytech/labels --depth 1 --main` should be enough.

Comment (alvicsam) [reply .github/workflows/check-labels.yml]: yeah, it checked out to a different branch initially, can be removed thx

File: .github/workflows/check-labels.yml [hunk 2]
```diff
@@ -31,11 +31,7 @@ jobs:
           echo "Cloning repo with labels specs"
 
           # Temporary, before https://github.com/paritytech/labels/pull/29 is not merged
-          git clone https://github.com/paritytech/labels
-          cd labels
-          git fetch origin main
-          git checkout main
-          cd ..
+          git clone https://github.com/paritytech/labels --depth 1 --main
```
Comment (ggwpez) [inline .github/workflows/check-labels.yml:34]:
```suggestion
          git clone https://github.com/paritytech/labels --depth 1 --branch main
```
My first comment was wrong, sorry.

File: .github/workflows/check-licenses.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      - uses: actions/setup-node@v3.8.1
+        with:
+          node-version: '18.x'
+          registry-url: 'https://npm.pkg.github.com'
+          scope: '@paritytech'
+      - name: Check the licenses
+        run: |
+          shopt -s globstar
+
+          npx @paritytech/license-scanner@0.0.5 scan \
```
Comment (ggwpez) [inline .github/workflows/check-licenses.yml:21]:
```suggestion
          npx --yes https://github.com/paritytech/license-scanner scan \
```

Not ideal, but maybe work as hot-fix.

**Review by ggwpez — APPROVED** [review]

**Review by liamaharon — APPROVED** [review]

**Review by oleg-plakida — APPROVED** [review]

**Review by rcny — APPROVED** [review]

**Pull request merged**